exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -42,6 +43,10 @@ func (r *AzureManagedMachinePool) Default(client client.Client) {
 	if r.Spec.Name == nil || *r.Spec.Name == "" {
 		r.Spec.Name = &r.Name
 	}
+
+	if r.Spec.OsDiskType == nil || to.String(r.Spec.OsDiskType) == "" {
```

Comment:
Could this be handled using a default comment on the type for OpenAPI to handle for us? (I'm not actually sure if that would handle both nil and empty string)

Reply:
I think we could add this:

`// +kubebuilder:default=Managed`

@CecileRobertMichon @zmalik is that preferable to enforcing in the `Default()` method of the webhook?

Reply:
Both are valid ways to do defaulting. I don't know if one is preferable over the other. OpenAPI defaulting is slightly cleaner since it doesn't require code in the webhook, but I don't know that it's a huge gain. It might be good to check if there are any API conventions out there that recommend one over the other (there isn't anything about this in https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md).

Reply:
I would prefer using the openapi solution, I'll test drive that out.

@zmalik do you have any thoughts?

Reply:
I also think that both ways are valid. And not sure about the standard preferred way. If it can be done through OpenAPI I would also prefer that way. 

> I'm not actually sure if that would handle both nil and empty string

from 1.20 we have this: 
```
When nullable is true, null values will be conserved and won't be defaulted
```

Reply:
Based on prior discussions about preferring an opinionated approach vs a passive one (let AKS choose the default for us), I think we should keep the current definition and not set `OsDiskType` to nullable.